build-tools/build.sh

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+#!/bin/sh
+
+buildToolsFolder="$(basename "$(dirname "$0")")"
+generatorFolder=$buildToolsFolder/../index/generator
+
+display_usage() { 
+  echo "A devfile registry repository folder must be passed in as an argument" 
+  echo "usage: ./build.sh <path-to-registry-repository-folder>" 
+} 
+
+# Check if a registry repository folder was passed in, if not, exit
+registryFolder=$1
+if [ $# -ne 1 ]; then
+  display_usage
+  exit 1
+fi
+
+# Build the index generator/validator
+echo "Building index-generator tool"
+cd $generatorFolder
```

Comment:
I think we can specify the generator build script and generator binary path at the beginning of this script as global variable, then we don't need to switch directory back and forth by using `cd <path>` + `cd "$OLDPWD"` way, it's also better for readability and maintainability I think.

Reply:
So, I tried to do that initially when I wrote the script, but the index generator build doesn't support being run outside of its own directory:
```
Johns-MacBook-Pro-3:registry-support johncollier$ index/generator/build.sh 
build flag -mod=vendor only valid when using modules
```

Reply:
Yeah, I think so, we have to inside the package dir to build go module, then sounds like we have to move the same thing (cd <path> + cd "$OLDPWD") to index generator build.sh. Given that I think we can keep the existing code.